crates/ide-completion/src/render.rs

```diff
@@ -147,7 +148,40 @@ pub(crate) fn render_field(
         .set_documentation(field.docs(db))
         .set_deprecated(is_deprecated)
         .lookup_by(name);
-    item.insert_text(field_with_receiver(db, receiver.as_ref(), &escaped_name));
+    if ty.is_fn() || ty.is_closure() {
+        let mut builder = TextEdit::builder();
+        // Using TextEdit, insert '(' before the struct name and ')' before the
+        // dot access, then comes the field name and optionally insert function
+        // call parens.
+
+        if let Some(receiver) = &dot_access.receiver {
+            let range = receiver.syntax().text_range();
+            builder.insert(range.start(), "(".to_string());
+            builder.insert(range.end(), ")".to_string());
+        }
```

Comment:
So that it works in macros too, you could use `ctx.completion.sema.original_ast_node()` like so:

```suggestion
        if let Some(receiver) = &dot_access.receiver {
            if let Some(receiver) = ctx.completion.sema.original_ast_node(receiver.clone()) {
                let range = receiver.syntax().text_range();
                builder.insert(range.start(), "(".to_string());
                builder.insert(range.end(), ")".to_string());
            }
        }
```